Assets/_BForBoss/_Core/Scripts/WaveSystem/WaveModel.cs

```diff
@@ -52,20 +54,25 @@ public void IncrementKillCount()
             KillCount++;
         }
 
+        public void UpdateKillCountView()
+        {
+            OnEnemyKillCountChanged?.Invoke(_maxEnemyCount - _killCount);
```

Comment:
So you have this separate invoke, because u need to return the inverse killCount. 

You can also remove `OnEnemyKillCountChanged` and just have both sides use `OnEnemyKilled`, and return `_maxEnemyCount - _killCount`.

And then in the `WaveManager`. use the logic `if (killCount <= 0)` instead.

Reply:
The back and forth call back logic is getting confusing. 
and the WaveManager now has a reference to a `View` of some sort, and that's coupling it even more.